script/watch.js

```diff
@@ -97,56 +100,124 @@ var liveInfo;
-			targetServer = (targetServer + 1) % servers.length;
-		}
-	});
+    $.getJSON("http://goalitium.kapsi.fi/dopelives_status3?callback=?", function (data) {
+        data = "a Game: Apex Legends https://www.youtube.com/watch?v=dYONccu0PZ8"
+        var streamURL = "";
+        var service = "";
+        var urlIndex = -1;
+        if ((urlIndex = data.indexOf("twitch.tv")) > 0) {
+            var startURL = Math.max(0,
+                data.lastIndexOf(" ", urlIndex) + 1,
+                data.lastIndexOf("\n", urlIndex) + 1,
+                data.lastIndexOf("(", urlIndex) + 1
+            );
+            var endURL = Math.min(data.length,
+                data.indexOf(" ", urlIndex) == -1 ? data.length : data.indexOf(" ", urlIndex),
+                data.indexOf("\n", urlIndex) == -1 ? data.length : data.indexOf("\n", urlIndex),
+                data.indexOf(")", urlIndex) == -1 ? data.length : data.indexOf(")", urlIndex)
+            );
+
+            streamURL = data.substring(startURL, endURL);
+            service = "twitch";
+        } else if ((urlIndex = data.indexOf("youtube.com")) > 0 || (urlIndex = data.indexOf("youtu.be")) > 0) {
+            var startURL = Math.max(0,
+                data.lastIndexOf(" ", urlIndex) + 1,
+                data.lastIndexOf("\n", urlIndex) + 1,
+                data.lastIndexOf("(", urlIndex) + 1
+            );
+            var endURL = Math.min(data.length,
+                data.indexOf(" ", urlIndex) == -1 ? data.length : data.indexOf(" ", urlIndex),
+                data.indexOf("\n", urlIndex) == -1 ? data.length : data.indexOf("\n", urlIndex),
+                data.indexOf(")", urlIndex) == -1 ? data.length : data.indexOf(")", urlIndex)
+            );
+
+            streamURL = data.substring(startURL, endURL);
+            service = "youtube";
+        }
+
+        var info = data.split("\n");
+
+        if (streamURL.length > 0 && validURL(streamURL)) {
+            if(service == "twitch" && !$("#twitch").is(":visible")) {
+                $("#twitch").show();
+                $("#youtube").hide();
+                $("#youtube").attr("src", "");
+                $("#flash").hide();
+                
+                var urlParts = streamURL.split("/");
+                var channelName = urlParts[urlParts.length-1] == "" ? urlParts[urlParts.length-2] : urlParts[urlParts.length-1];
+
+                $("#twitch").attr("src","https://player.twitch.tv/?channel=" + channelName);
+            }
+            else if(service == "youtube" && !$("#youtube").is(":visible")) {
+                $("#twitch").hide();
+                $("#twitch").attr("src","");
+                $("#youtube").show();
+                $("#flash").hide();
+
+                var urlParts = streamURL.split("?v=");
+                var channelId = urlParts[urlParts.length - 1];
+
+                $("#youtube").attr("src", "https://www.youtube.com/embed/" + channelId + "?autoplay=1");
+            }
+        }
+        else {

```

Comment:
Are we not missing out on the `updateInfoPane()` calls from that show the streamer/game in UI now?

Reply:
Twitch and Youtube provide their own titles, I think having our title displaying on top of that will only be confusing.

Reply:
Ah right, if they're in the respective players already then that makes sense indeed.